src/ck-core/ckrdma.C

```diff
@@ -1883,15 +1867,20 @@ void sendAckMsgToParent(envelope *env)  {
 }
 
 CkArray* getArrayMgrFromMsg(envelope *env) {
+
+  env->setMsgtype(ForArrayEltMsg); // Used to bypass assertion in getArrayMgr
```

Comment:
Same as above. I think the group id in the envelope can be used instead of array mgr.

Reply:
Similarly, getArrayMgrFromMsg is only called either from the pe performing the `Rget` or from the intermediate node in the bcast tree. Even there, it looks like only one PE actually calls it. However, even this can be replaced with your suggestion of using the group id. I'll make that change and push.